tests/unit/test__job_helpers.py

```diff
@@ -605,6 +601,7 @@ def test_query_and_wait_sets_job_creation_mode(monkeypatch: pytest.MonkeyPatch):
 
 def test_query_and_wait_sets_location():
     client = mock.create_autospec(Client)
+    client.default_job_creation_mode = None
```

Comment:
Just for my education, why do we need to set `client.default_job_creation_mode = None` explicitly? It seems using default value would be enough.

Reply:
Originally I tried to keep the asserted request unchanged, but I've instead made this less confusing and used mock.ANY for the jobCreationMode aspect of the request.